Source/cmake_core.cmake

```diff
@@ -136,68 +142,72 @@ macro(astcenc_set_properties NAME)
+                    ASTCENC_NEON=0
+                    ASTCENC_SSE=0
+                    ASTCENC_AVX=0
+                    ASTCENC_POPCNT=0
+                    ASTCENC_F16C=0)
+
+        elseif(${ISA_SIMD} MATCHES "neon")
+            target_compile_definitions(${NAME}
+                PRIVATE
+                    ASTCENC_NEON=1
+                    ASTCENC_SSE=0
+                    ASTCENC_AVX=0
+                    ASTCENC_POPCNT=0
+                    ASTCENC_F16C=0)
+
+        elseif(${ISA_SIMD} MATCHES "sse2")
+            target_compile_definitions(${NAME}
+                PRIVATE
+                    ASTCENC_NEON=0
+                    ASTCENC_SSE=20
+                    ASTCENC_AVX=0
+                    ASTCENC_POPCNT=0
+                    ASTCENC_F16C=0)
+
+        elseif(${ISA_SIMD} MATCHES "sse4.1")
+            target_compile_definitions(${NAME}
+                PRIVATE
+                    ASTCENC_NEON=0
+                    ASTCENC_SSE=41
+                    ASTCENC_AVX=0
+                    ASTCENC_POPCNT=1
+                    ASTCENC_F16C=0)
+
+        elseif(${ISA_SIMD} MATCHES "avx2")
+            target_compile_definitions(${NAME}
+                PRIVATE
+                    ASTCENC_NEON=0
+                    ASTCENC_SSE=41
+                    ASTCENC_AVX=2
+                    ASTCENC_POPCNT=1
+                    ASTCENC_F16C=1)
+        endif()
+    endif()
 
-    elseif(${ISA_SIMD} MATCHES "sse2")
-        target_compile_definitions(${NAME}
+    if(${ISA_SIMD} MATCHES "avx2")
+        target_compile_options(${NAME}
             PRIVATE
-                ASTCENC_NEON=0
-                ASTCENC_SSE=20
-                ASTCENC_AVX=0
-                ASTCENC_POPCNT=0
-                ASTCENC_F16C=0)
+                $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-mavx2 -mpopcnt -mf16c>
+                $<$<CXX_COMPILER_ID:MSVC>:/arch:AVX2>)
 
     elseif(${ISA_SIMD} MATCHES "sse4.1")
-        target_compile_definitions(${NAME}
-            PRIVATE
-                ASTCENC_NEON=0
-                ASTCENC_SSE=41
-                ASTCENC_AVX=0
-                ASTCENC_POPCNT=1
-                ASTCENC_F16C=0)
-
         target_compile_options(${NAME}
             PRIVATE
                 $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-msse4.1 -mpopcnt>)
 
-    elseif(${ISA_SIMD} MATCHES "avx2")
-        target_compile_definitions(${NAME}
-            PRIVATE
-                ASTCENC_NEON=0
-                ASTCENC_SSE=41
-                ASTCENC_AVX=2
-                ASTCENC_POPCNT=1
-                ASTCENC_F16C=1)
-
-        target_compile_options(${NAME}
-            PRIVATE
-                $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-mavx2 -mpopcnt -mf16c>
-                $<$<CXX_COMPILER_ID:MSVC>:/arch:AVX2>)
-
     elseif(${ISA_SIMD} MATCHES "native")
         target_compile_options(${NAME}
             PRIVATE
-                $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-march=native>)
+                $<$<CXX_COMPILER_ID:GNU>: -march=native>)
```

Comment:
This breaks clang on Linux.

Reply:
Fixed in https://github.com/ARM-software/astc-encoder/pull/266/commits/1ef2b84a31901c36f85479527cfe48d8365546a1